### PR TITLE
hide contact info for unauthorized users

### PR DESCRIPTION
**Pull request by Mh-Asmi**
This pull request makes the following changes:
- hide author and contact info for users don't have a manage-post permission
- confirm that this info contact info will not return inside message too
- apply these rules on `api/v5/posts` and `api/v5/posts/{id}` 

Test checklist:
- [ ]

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

File: src/Ushahidi/Modules/V5/Actions/Post/Handlers/AbstractPostQueryHandler.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        if ($user->role === "admin") {
+            return true;
+        }
+        $permissions =
+            RolePermission::select("permission")->where('role', '=', $user->role)->get()->pluck('permission');
+        if (in_array("Manage Posts", $permissions->toArray())) {
+            return true;
+        }
+        return false;
```
Comment (webong) [inline src/Ushahidi/Modules/V5/Actions/Post/Handlers/AbstractPostQueryHandler.php:27]: This piece of code already exists in Ushahidi\Core\Tool\Acl class so no need for code duplication

**Review by webong — APPROVED** [review]
I'll give this a good pass since this is a hotfix. We can always refactor in future iteration

File: src/Ushahidi/Modules/V5/Actions/Post/Handlers/AbstractPostQueryHandler.php (hunk 1 above)
Comment (webong) [reply src/Ushahidi/Modules/V5/Actions/Post/Handlers/AbstractPostQueryHandler.php:27]: See here https://github.com/ushahidi/platform/blob/88c5381da67bf95bd28dc9f637282308a2357bcf/src/Ushahidi/Core/Tool/Acl.php#L36-L56

So the best way would be to inject the Acl class via the constructor into the handler, Laravel Container handles the resolution, and we can the do permission checks

`$this->accessControl->hasPermission($user, 'Manage Post');`

**Comment by tuxpiper** [comment]
> I'll give this a good pass since this is a hotfix. We can always refactor in future iteration

Thank you @webong 🙏  Could you please create a quick technical task in Linear, so that we don't forget?

**Pull request merged**